pkg/column-mapping/column.go

```diff
@@ -302,23 +366,119 @@ func tableName(schema, table string) string {
 	return fmt.Sprintf("`%s`.`%s`", schema, table)
 }
 
-func addPrefix(info *columnInfo, vals []interface{}) []interface{} {
+func addPrefix(info *mappingInfo, vals []interface{}) ([]interface{}, error) {
 	prefix := info.rule.Arguments[0]
-	vals[info.targetPosition] = fmt.Sprintf("%s:%v", prefix, vals[info.targetPosition])
-	return vals
+	originStr, ok := vals[info.targetPosition].(string)
+	if !ok {
+		return nil, errors.NotValidf("column %d value is not string, but %v", info.targetPosition, vals[info.targetPosition])
+	}
+
+	// fast to concatenated string
+	rawByte := make([]byte, 0, len(prefix)+len(originStr))
+	rawByte = append(rawByte, prefix...)
+	rawByte = append(rawByte, originStr...)
+
+	vals[info.targetPosition] = string(rawByte)
+	return vals, nil
 }
 
-func addSuffix(info *columnInfo, vals []interface{}) []interface{} {
+func addSuffix(info *mappingInfo, vals []interface{}) ([]interface{}, error) {
 	suffix := info.rule.Arguments[0]
-	vals[info.targetPosition] = fmt.Sprintf("%v:%s", vals[info.targetPosition], suffix)
-	return vals
+	originStr, ok := vals[info.targetPosition].(string)
+	if !ok {
+		return nil, errors.NotValidf("column %d value is not string, but %v", info.targetPosition, vals[info.targetPosition])
+	}
+
+	rawByte := make([]byte, 0, len(suffix)+len(originStr))
+	rawByte = append(rawByte, originStr...)
+	rawByte = append(rawByte, suffix...)
+
+	vals[info.targetPosition] = string(rawByte)
+	return vals, nil
+}
+
+func partitionID(info *mappingInfo, vals []interface{}) ([]interface{}, error) {
+	// only int64 now
+	var (
+		originID int64
+		err      error
+		isChars  bool
+	)
+
+	switch rawID := vals[info.targetPosition].(type) {
+	case int:
+		originID = int64(rawID)
+	case int8:
+		originID = int64(rawID)
+	case int32:
+		originID = int64(rawID)
+	case int64:
+		originID = rawID
+	case uint:
+		originID = int64(rawID)
+	case uint16:
+		originID = int64(rawID)
+	case uint32:
+		originID = int64(rawID)
+	case uint64:
+		originID = int64(rawID)
+	case string:
+		originID, err = strconv.ParseInt(rawID, 10, 64)
+		if err != nil {
+			return nil, errors.NotValidf("column %d value is not int, but %v", info.targetPosition, vals[info.targetPosition])
+		}
+		isChars = true
+	}
+
+	if originID >= maxOriginID || originID < 0 {
+		return nil, errors.NotValidf("id must less than %d, bigger or equal than 0, but get %d", maxOriginID, originID)
+	}
+
+	originID = int64(info.instanceID | info.schemaID | info.tableID | originID)
+	if isChars {
+		vals[info.targetPosition] = strconv.FormatInt(originID, 10)
+	} else {
+		vals[info.targetPosition] = originID
+	}
+
+	return vals, nil
 }
 
-func cloneColumn(info *columnInfo, vals []interface{}) []interface{} {
-	newVals := make([]interface{}, 0, len(vals)+1)
-	newVals = append(newVals, vals[:info.targetPosition]...)
-	newVals = append(newVals, vals[info.sourcePosition])
```

Comment:
seems `sourcePosition` is not used anymore now. keep it for future usage?

Reply:
yep